tools/RAiDER/getStationDelays.py

```diff
@@ -0,0 +1,184 @@
+                       %(doy,doyFromFile,j))
+                   continue
+                d.append(trotot)
+                ngrad.append(tgntot)
+                egrad.append(tgetot)
+                timesList.append(seconds)
+                Sig.append(trototSD)
+            if 'TROP/SOLUTION' in line:
+                flag = True
+        del f
+        # Break iteration if file contains no data.
+        if d == []:
+            print('WARNING: file %s for station %s is empty, will continue reading next tarfile(s)'%(j, j.split('.')[0]))
+            continue
+
+        # check for missing times
+        true_times = list(range(0,86400, 300))
+        if len(timesList)!=len(true_times):
+           missing = [True if t not in timesList else False for t in true_times]
+           mask = np.array(missing)
+           delay, sig, east_grad, north_grad = [np.full((288,), np.nan)]*4 
+           delay[~mask] = d
+           sig[~mask] = Sig
+           east_grad[~mask] = egrad
+           north_grad[~mask] = ngrad
+           times = true_times.copy()
+        else:
+           delay     = np.array(d)
+           times     = np.array(timesList)
+           sig       = np.array(Sig)
+           east_grad = np.array(egrad)
+           north_grad= np.array(ngrad)
+
+        # if time not specified, pass all times
+        if returnTime == None:
+            filtoutput = {'ID': [site]*len(north_grad), 'Date': [time]*len(north_grad), 'ZTD': delay, 'north_grad': north_grad, \
+                'east_grad': east_grad, 'doy': times, 'sigZTD': sig}
+            filtoutput = [{key : value[k] for key, value in filtoutput.items()} for k in range(len(filtoutput['ID']))]
+        else:
+            index = np.argmin(np.abs(np.array(timesList) - returnTime))
+            filtoutput = [{'ID': site, 'Date': time, 'ZTD': delay[index], 'north_grad': north_grad[index], \
+                'east_grad': east_grad[index], 'doy': times[index], 'sigZTD': sig[index]}]
+        # setup pandas array and write output to CSV, making sure to update existing CSV.
+        filtoutput = pd.DataFrame(filtoutput)
+        if not os.path.exists(filename):
+            filtoutput.to_csv(filename, index=False)
+        else:
+            filtoutput.to_csv(filename, index=False, mode='a', header=False)
+
+    # record all used tar files
+    allstationTarfiles.extend([os.path.join(stationFile,k) for k in stationTarlist])
+    allstationTarfiles.sort()
+    del ziprepo
+
+    return allstationTarfiles
+
+
+def getStationData(inFile, outDir = None, returnTime = None):
+    '''
+    Pull tropospheric delay data for a given station name
+    '''
+
+    if outDir is None:
+        outDir = os.getcwd()
+
+    pathbase = os.path.join(outDir, 'GPS_delays')
+    if not os.path.exists(pathbase):
+        os.mkdir(pathbase)
+
+    returnTime=(int(returnTime.split(':')[0])*3600)+(int(returnTime.split(':')[1])*60)+int(returnTime.split(':')[2])
```

Comment:
Might be a little cleaner to call this part that parses the returnTime as its own function

Reply:
Lots of ways to do this. Definitely extract it to a function though. Probably the most straight forward is:
```suggestion
    h, m, s = returnTime.split(":")
    returnTime = int(h * 3600 + m * 60 + s)
```

Reply:
Make sense, moved out to a separate function